xblock/runtime.py

```diff
@@ -1211,10 +1212,26 @@ class Mixologist(object):
     """
     def __init__(self, mixins):
         """
-        :param mixins: Classes to mixin
+        :param mixins: Classes to mixin or names of classes to mixin.
         :type mixins: `iterable` of `class`
         """
-        self._mixins = tuple(mixins)
+
+        mixin_classes = []
+        for mixin in mixins:
+            if isinstance(mixin, six.text_type):
+                module, cls = mixin.rsplit('.', 1)
+                imported_module = importlib.import_module(module)
+                try:
+                    mixin_class = getattr(imported_module, cls)
+                except AttributeError as e:
+                    msg = "Couldn't import class {}: {}".format(mixin, str(e))
```

Comment:
This try-clause should include lines 1222 and 1223 also, since they can also indicate problems with the class name.
Also: no need for `str(e)`, `e` would give the same result, since `{}` will str it for you.

If you expand the scope of the except, then we don't know what exception to catch, and you'd want to include the class in the message:
```
except Exception as e:
    msg = "Couldn't import class {!r}: {}: {}".format(mixin, e.__class__.__name__, e)
    raise ImportError(msg)
```

Reply:
sounds good, I'll make those changes thanks!